Source/FieldSolver/ImplicitSolvers/ThetaImplicitEM.H

```diff
@@ -0,0 +1,142 @@
+/* Copyright 2024 Justin Angus
+ *
+ * This file is part of WarpX.
+ *
+ * License: BSD-3-Clause-LBNL
+ */
+#ifndef THETA_IMPLICIT_EM_H_
+#define THETA_IMPLICIT_EM_H_
+
+#include "FieldSolver/ImplicitSolvers/WarpXSolverVec.H"
+#include "NonlinearSolvers/NonlinearSolverLibrary.H"
+
+#include <AMReX_Array.H>
+#include <AMReX_MultiFab.H>
+#include <AMReX_REAL.H>
+
+#include "ImplicitSolver.H"
+
+/** @file
+ *  Theta-implicit electromagnetic time solver class. This is a fully implicit
+ *  algorithm where both the fields and particles are treated implicitly.
+ *
+ *  The time stencil is
+ *  Eg^{n+1} = Eg^n + c^2*dt*( curlBg^{n+theta} - mu0*Jg^{n+1/2} )
+ *  Bg^{n+1} = Bg^n - dt*curlEg^{n+theta}
+ *  xp^{n+1} = xp^n + dt*up^{n+1/2}/(0.5*(gammap^n + gammap^{n+1}))
```

Comment:
This is equivalent, but slightly shorter
```suggestion
 *  xp^{n+1} = xp^n + dt*up^{n+1/2}/gammap^{n+1/2}
```